front/components/skill_builder/SkillBuilder.tsx

```diff
@@ -0,0 +1,222 @@
+import {
+  BarFooter,
+  BarHeader,
+  Button,
+  cn,
+  Input,
+  ScrollArea,
+  TextArea,
+  XMarkIcon,
+} from "@dust-tt/sparkle";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useRouter } from "next/router";
+import { useState } from "react";
+import { useController, useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { appLayoutBack } from "@app/components/sparkle/AppContentLayout";
+import { useSendNotification } from "@app/hooks/useNotification";
+import type { UserType, WorkspaceType } from "@app/types";
+
+const skillBuilderFormSchema = z.object({
+  name: z
+    .string()
+    .min(1, "Skill name is required")
+    .refine((value) => !/\s/.test(value), "Skill name cannot contain spaces"),
+  description: z.string().min(1, "Skill description is required"),
+  instructions: z.string().min(1, "Skill instructions are required"),
+});
+
+type SkillBuilderFormData = z.infer<typeof skillBuilderFormSchema>;
+
+interface SkillBuilderProps {
+  owner: WorkspaceType;
+  user: UserType;
+}
+
+export default function SkillBuilder({ owner }: SkillBuilderProps) {
+  const router = useRouter();
+  const sendNotification = useSendNotification();
+  const [isSaving, setIsSaving] = useState(false);
+
+  const form = useForm<SkillBuilderFormData>({
+    resolver: zodResolver(skillBuilderFormSchema),
+    defaultValues: {
+      name: "",
+      description: "",
+      instructions: "",
+    },
+  });
+
+  const { field: nameField, fieldState: nameFieldState } =
```

Comment:
Could we isolate each component in a dedicated component?
Rationale: we are gonna end up doing more sophisticated things, probably requiring to watch some fields, and take actions based on it.. Having all the logic in one component is gonna become hardly readable